mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentController.java

```diff
@@ -646,16 +690,77 @@ public View onCreateView(@NonNull LayoutInflater inflater,
                 watermark.setVisibility(View.GONE);
             }
 
-            final Button button = view.findViewById(R.id.button);
-            int buttonTextResId = arguments.getInt("buttontext");
+            if (initButton(arguments, view, R.id.button1, KEY_BUTTON_1_TEXT, KEY_BUTTON_1_TAG)
+                    & initButton(arguments, view, R.id.button2, KEY_BUTTON_2_TEXT,
+                    KEY_BUTTON_2_TAG)) {
+                needsOr = true;
+            }
+
+            TextView or = view.findViewById(R.id.or);
+            or.setVisibility(needsOr ? View.VISIBLE : View.GONE);
+
+            return view;
+        }
+
+        @Override
+        public void onClick(View view) {
+            if (view.getTag().equals(BUTTON_TAG_OPEN_SETTINGS)) {
+                Intent preferencesIntent = new Intent(getActivity(), PreferencesActivity.class);
+                TaskStackBuilder.create(getActivity())
+                        .addNextIntentWithParentStack(preferencesIntent)
+                        .startActivities();
+            } else if (view.getTag().equals(BUTTON_TAG_ENABLE_DEMO_MODE)) {
+                PreferenceManager.getDefaultSharedPreferences(getContext())
+                        .edit()
+                        .putBoolean(Constants.PREFERENCE_DEMOMODE, true)
+                        .apply();
+            } else if (view.getTag().equals(BUTTON_TAG_ENABLE_WIFI)) {
+                WifiManager wifiManager =
+                        (WifiManager) getContext().getSystemService(Context.WIFI_SERVICE);
+                wifiManager.setWifiEnabled(true);
+            } else if (view.getTag().equals(BUTTON_TAG_RETRY_NETWORK)) {
+                ConnectionFactory.restartNetworkCheck();
+                getActivity().recreate();
+            } else if (view.getTag().equals(BUTTON_TAG_RETRY_SERVER_PROP_FETCH)) {
+                ((MainActivity) getActivity()).retryServerPropertyQuery();
+            }
+        }
```

Comment:
Could use a lambda for each of the cases, assign the lambda as tag and execute it in onClick. Would save the indirection.

Reply:
Some actions are used more than once, e.g. enable wifi and open settings. This is why I put everything in this function.